Common/ColorConv.cpp

```diff
@@ -322,13 +322,12 @@ void ConvertRGBA565ToRGBA8888(u32 *dst32, const u16 *src, u32 numPixels) {
 	u32 i = 0;
 #endif
 
-	u8 *dst = (u8 *)dst32;
 	for (u32 x = i; x < numPixels; x++) {
 		u16 col = src[x];
-		dst[x * 4] = Convert5To8((col) & 0x1f);
-		dst[x * 4 + 1] = Convert6To8((col >> 5) & 0x3f);
-		dst[x * 4 + 2] = Convert5To8((col >> 11) & 0x1f);
-		dst[x * 4 + 3] = 255;
+		dst32[x] = Convert5To8((col) & 0x1f);
+		dst32[x] |= Convert6To8((col >> 5) & 0x3f) << 8;
+		dst32[x] |= Convert5To8((col >> 11) & 0x1f) << 16;
+		dst32[x] |= 255 << 24;
```

Comment:
Might be better to use a temporary for all these.  I think we may use these funcs sometimes to write directly to uncached GPU buffers...

-[Unknown]

Reply:
oh ok.
I think it would still be optimized into a single write, since the pointer isn't marked volatile, but I'll change that into a temporary.